tests/api/common/test_airflow_health.py

```diff
@@ -17,24 +17,26 @@
 from __future__ import annotations
 
 from datetime import datetime
-from unittest.mock import MagicMock
+from unittest.mock import MagicMock, patch
+
+import pytest
 
 from airflow.api.common.airflow_health import (
     HEALTHY,
     UNHEALTHY,
-    DagProcessorJobRunner,
-    SchedulerJobRunner,
-    TriggererJobRunner,
     get_airflow_health,
 )
 
+pytestmark = pytest.mark.db_test
 
-def test_get_airflow_health_only_metadatabase_healthy():
-    SchedulerJobRunner.most_recent_job = MagicMock(return_value=None)
-    TriggererJobRunner.most_recent_job = MagicMock(return_value=None)
-    DagProcessorJobRunner.most_recent_job = MagicMock(return_value=None)
-    health_status = get_airflow_health()
 
+@patch("airflow.api.common.airflow_health.SchedulerJobRunner.most_recent_job", return_value=None)
```

Comment:
These test had bad side-effects on other tests because most_recent_job remained patched with MagicMock after the tests completed.